hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsHttpOperation.java

```diff
@@ -161,6 +163,44 @@ public String toString() {
     return sb.toString();
   }
 
+  // Returns a trace message for the ABFS API logging service to consume
+  public String toKvpString() {
+    String urlStr = null;
+
+    try{
+      urlStr = URLEncoder.encode(url.toString(), "UTF-8");
+    } catch(UnsupportedEncodingException e) {
+      urlStr = "https%3A%2F%2Ffailed%2Fto%2Fencode%2Furl";
+    }
+
+    final StringBuilder sb = new StringBuilder();
+    sb.append("s=");
```

Comment:
do:
final StringBuilder sb = new StringBuilder()
.append("s=")
.append(statusCode)
.append(" e=")
....

Reply:
Did this.